utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala

```diff
@@ -0,0 +1,135 @@
+  def merge[K, V](@ParamName("map1") map1: java.util.Map[K, V], @ParamName("map2") map2: java.util.Map[K, V]): java.util.Map[K, V] = {
+    val merged = new java.util.LinkedHashMap[K, V](map1)
+    merged.putAll(map2)
+    merged
+  }
+
+  @Documentation(description = "Returns the smallest element (elements must be comparable)")
+  @GenericType(typingFunction = classOf[TypingFunctions.ListElementTyping])
+  def min[T<:Comparable[T]](@ParamName("list") list: java.util.Collection[T]): T = {
+    checkIfComparable(list)
+    Collections.min[T](list)
+  }
+
+  @Documentation(description = "Returns the largest element (elements must be comparable)")
+  @GenericType(typingFunction = classOf[TypingFunctions.ListElementTyping])
+  def max[T<:Comparable[T]](@ParamName("list") list: java.util.Collection[T]): T = {
+    checkIfComparable(list)
+    Collections.max[T](list)
+  }
+
+  @Documentation(description = "Returns a slice of the list starting with start index (inclusive) and ending at stop index (exclusive)")
+  @GenericType(typingFunction = classOf[TypingFunctions.ListTyping])
+  def slice[T](@ParamName("list") list: java.util.Collection[T], @ParamName("start") start: Int, @ParamName("stop") stop: Int): java.util.List[T] = {
+    list.asScala.slice(start, stop).toList.asJava
+  }
+
+  @Documentation(description = "Returns a sum of all elements")
+  def sum[T <: java.lang.Number](@ParamName("listOfNumbers") list: java.util.Collection[T]): java.lang.Number = {
+    list.asScala.map(x => new java.math.BigDecimal(x.toString)).reduce((a, b) => a.add(b))
+  }
+
+  @Documentation(description = "Returns a list of all elements sorted in ascending order (elements must be comparable)")
+  @GenericType(typingFunction = classOf[TypingFunctions.ListTyping])
+  def sorted[T<:Comparable[T]](@ParamName("list") list: java.util.Collection[T]): java.util.List[T] = {
+    checkIfComparable(list)
+    val values = new java.util.ArrayList[T](list)
+    Collections.sort[T](values)
+    values
+  }
+
+  @GenericType(typingFunction = classOf[TypingFunctions.MapTyping])
+  @Documentation(description = "Returns a map with elements sorted by value in ascending order")
+  def sortedMapByValueAsc[K, V<:Comparable[V]](@ParamName("map") map: java.util.Map[K, V]): java.util.Map[K, V]
+  = ListMap(map.asScala.toSeq.sortBy(_._2):_*).asJava
+
+  @GenericType(typingFunction = classOf[TypingFunctions.MapTyping])
+  @Documentation(description = "Returns a map with elements sorted by value in descending order")
+  def sortedMapByValueDesc[K, V<:Comparable[V]](map: java.util.Map[K, V]): java.util.Map[K, V]
+  = ListMap(map.asScala.toSeq.sortBy(_._2)(implicitly[Ordering[V]].reverse):_*).asJava
+
+  @Documentation(description = "Returns a list made of first n elements of the given list")
+  @GenericType(typingFunction = classOf[TypingFunctions.ListTyping])
+  def take[T](@ParamName("list") list: java.util.List[T], @ParamName("max") max: Int): java.util.List[T]
```

Comment:
Maybe we could have takeLeft, takeRight, dropLeft, dropRight. WDYT?

Reply:
Would a non-programmer understand what are "left" and "right" items in a collection? IMO "first"/"last" are better, and I'm not sure "first" is even needed. IMO take/takeLast and drop/dropLast reads better than takeLeft/takeRight and dropLeft/dropRight.

Reply:
Makes sense. Let's stick to Scala/Kotlin naming convention